### PR TITLE
[UXP-22637][UXP-22636] - Support for underlay and modal for unlocking sp-dialog usages

### PR DESCRIPTION
**Pull request by inasirhussain**
Fixes - 
https://jira.corp.adobe.com/browse/UXP-22636
https://jira.corp.adobe.com/browse/UXP-22637

**uxp-underlay.css** has below css property to polyfill **inset** css as its not supported in UXP
:host {
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
}

**uxp-modal.css** has below css property to polyfill **inset** css as its not supported in UXP
.fullscreenTakeover {
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
}

**uxp-dialog-base.css** has got following css as SWC relies on **dvh (Dynamic Viewport Height)** This is a newer unit introduced to address the issue with vh but dvh is not supported in UXP.
:host {
    height: 100vh;
}

### CODE REVIEW

File: packages/underlay/src/Underlay.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    constructor() {
+        super(...arguments);
+        // Add event listener for Escape key
+        this.addEventListener('keydown', this.handleEscape);
+    }
+
+    handleEscape(event) {
+        if (event.key === 'Escape') {
```
Comment (deepthi-majji) [inline packages/underlay/src/Underlay.js]: Why is it required?

File: packages/underlay/src/Underlay.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            console.log('handleEscape');
```
Comment (deepthi-majji) [inline packages/underlay/src/Underlay.js]: Remove log.

File: packages/dialog/src/uxp-dialog-base.css [hunk 3]
```diff
@@ -11,3 +11,7 @@ governing permissions and limitations under the License.
 */
 
 /* write uxp style overrides */
+
+:host {
+    height: 100vh;
```
Comment (deepthi-majji) [inline packages/dialog/src/uxp-dialog-base.css:16]: Can you please add one line comment about each css override in the PR description?

File: projects/swc-starter-webpack/src/example_usages/sp-underlay.html [hunk 4]
```diff
@@ -0,0 +1,653 @@
...
+                    />
```
Comment (deepthi-majji) [inline projects/swc-starter-webpack/src/example_usages/sp-underlay.html:332]: the src is too long

File: projects/swc-starter-webpack/src/styles.css [hunk 5]
```diff
@@ -89,12 +89,18 @@ sp-action-group {
 }
 
 sp-textfield,
+sp-tabs,
```
Comment (deepthi-majji) [inline projects/swc-starter-webpack/src/styles.css]: sp-tabs has margin-bottom: 40px below.

File: packages/underlay/src/Underlay.js (hunk 1 above)
Comment (inasirhussain) [reply packages/underlay/src/Underlay.js]: Missed to remove it, had added to test in the last moment. Fixed by removing it

File: packages/underlay/src/Underlay.js (hunk 2 above)
Comment (inasirhussain) [reply packages/underlay/src/Underlay.js]: Removed

File: projects/swc-starter-webpack/src/styles.css (hunk 5 above)
Comment (inasirhussain) [reply projects/swc-starter-webpack/src/styles.css]: Kept 40px one.

File: projects/swc-starter-webpack/src/example_usages/sp-underlay.html (hunk 4 above)
Comment (inasirhussain) [reply projects/swc-starter-webpack/src/example_usages/sp-underlay.html:332]: replaced with a shorter one.

File: projects/swc-starter-webpack/src/example_usages/sp-underlay.html [hunk 6]
```diff
@@ -0,0 +1,653 @@
...
...WuJmJ5whpqvbibIKbNw4wx70b5ojiQ42gDBC9qGVsOEUKnnjLndvIAHBkIP6UWJPurOot/hf/WXzSMz+ZvIA2kdjTTe3t8JwtvcIkSjnI5zS7rkYop8EqtLpUJ2Ab85+dO1pcMQecMPb0qEaLFeteM8127xH0Cng1JI7Bd7MoLMB3Pc0+ErFSgkOB1cRjElxEBnHPGKSPq8Llj96jC+4NNV5vgBCWyFmOOaSx283wHyRtGdyn1onIiOMX3et28fAuUf5DvSU6vFMzlZnXcMYOaTXMKOwIhVGXt71jyhI8SBzgZG1KCw9tCfWtWEbQRostzkjIHoKygvfrKrNJAy4GMhMHvWVB1DdDrS/dgkLxFwDklMf1rU1zPcKA10sYx3TApqtXDgEonA9qV2saSRbii5GfSmOQKikH2YtSSjXzFgc7Sw5/OnGCW27C5Gf/lxSGK0hSZSIlyygk4pQkUapgRJ3x2oV0FSTFaJH6TYOM5zxSSC2nkmMpnRgBhMN6Vk6Jt27FxigwsFO0KoGPQV34OoPklmZwrs7Njjn+lR3qjoS36guFufvk9jcoQTt+IMR7in0cgt2I7YNalZpXO5ie3rUxk0DQl0TRY9Ih2IzSErtZ3XAHfsKXB4YwI44nlPox7E0XcztbRbl+Ijtv59DSYXc0wUmVlDEZVeBUPl2HEWpbzSIcwogJwcnmhacohmaKTy0EYwu5s5pIkQd2JLEqeMnNLIItztliaXQQfNdQ43OYlQHJJFFP5LI0kRC3CcrIU9R2pROwtLbKor7f8AeM0yXmu3TzmH4FQuM7VriUONv1FFqgiQALPglx5ZO1hW36ge2huI7lXyEJDCA4OKiouni1FZY8RvuydoxnmpNc3kptpdzBwV7MoIqU+CHHkV6V1FZahp6Lb+duAG4OoBX5UOe9E6ujRy/AMjGBUQ05GdG/muh3d1IBpVGZVwouJueM7uaUwoofINUu5XYwhxCDhlLAEVI7G9uJY+NynOCrSDkVXUqusUrCaXepxu3cn60kiu5sKxldjvI5Y0Kk4h7Uy0JUnmkzsLD38wUWzSodhU/wD5BURs7mR05Y9wO5+dGwSMb8xk5XHqaJTt0DtpWPU8bKdxjPGc/wAwHNJwZJshY2x7hhQmhRouVHai4kCICB+9MXAu7BKssW5GRmjx6uKykeoxL5R754Oc1lCmyej/2Q=="
```
Comment (deepthi-majji) [inline projects/swc-starter-webpack/src/example_usages/sp-underlay.html]: Can you replace this src as well?

**Review by deepthi-majji — APPROVED** [review]

**Pull request merged**